src/Storages/ExternalStream/Kafka/WriteBufferFromKafka.cpp

```diff
@@ -20,15 +21,21 @@ void WriteBufferFromKafka::nextImpl()
     if (!offset())
         return;
 
+    /// `write_to_topic` should have been called before the buffer starts to work.
+    assert(topic);
+
+    auto * part_id = new Int32(partition_id);
```

Comment:
We don't need allocate, just convert partition_id to a `void *` , like 

```
auto err = rd_kafka_produce(
    ...
    reinterpret_cast<void *>(part_id) /* opaque */);
```

And then convert it back in KafkaSink

```
auto partition_id = reinterpret_cast<Int32>(opaque)
```